integration-tests/artifacts/installer/v1alpha1/installer.go

```diff
@@ -0,0 +1,145 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	provider "github.com/openebs/maya/pkg/provider/v1alpha1"
+	unstruct "github.com/openebs/maya/pkg/unstruct/v1alpha1"
+
+	unstructuredUtil "github.com/openebs/maya/pkg/util"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// Installer abstracts the installation/deletion of a
+// resource
+type Installer struct {
+	object     *unstructured.Unstructured
+	predicates []Predicate
+}
+
+// Predicate is an abstracts the verification of
+// a resource
+type Predicate func(*unstructured.Unstructured) bool
+
+// String returns the meta information of the
+// installing resource
+func (i *Installer) String() string {
+	if i.object == nil {
+		return "installer - object=nil"
+	}
+	return fmt.Sprintf("installer - name=%s namespace=%s kind=%s",
+		i.object.GetName(),
+		i.object.GetNamespace(),
+		i.object.GetKind(),
+	)
+}
+
+// GoString returns the go string of the installer
+func (i *Installer) GoString() string {
```

Comment:
can it be GetMetaInfo()?

Reply:
cc @AmitKumarDas

Reply:
It has to be GoString & String. They implement Stringer & GoStringer interfaces